TeslaMateAgile/Services/NordpoolService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Microsoft.Extensions.Options;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using TeslaMateAgile.Data;
+using TeslaMateAgile.Data.Options;
+using TeslaMateAgile.Services.Interfaces;
+
+namespace TeslaMateAgile.Services;
+
+public class NordpoolService : IPriceDataService
+{
+    private readonly HttpClient _client;
+    private readonly NordpoolOptions _options;
+
+    public NordpoolService(HttpClient client, IOptions<NordpoolOptions> options)
+    {
+        _client = client;
+        _options = options.Value;
+    }
+
+    public async Task<IEnumerable<Price>> GetPriceData(DateTimeOffset from, DateTimeOffset to)
+    {
+        var url = "?currency=," + _options.Currency + "&endDate=" + to.UtcDateTime.ToString("dd-MM-yyyy");
+
+        var resp = await _client.GetAsync(url);
+
+        resp.EnsureSuccessStatusCode();
+
+        var prices = new List<Price>();
+
+        var nordPoolResponse = await JsonSerializer.DeserializeAsync<NordpoolResponse>(await resp.Content.ReadAsStreamAsync());
+
+        if (nordPoolResponse.data.Rows.Count > 0)
+        {
+            foreach (var row in nordPoolResponse.data.Rows)
+            {
+                if (row.IsExtraRow)
+                {
+                    continue;
+                }
+
+                foreach (var column in row.Columns)
+                {
+                    decimal.TryParse(column.Value.Replace(',', '.'), out var value);
```

Comment:
What happens if this parse fails? Is this likely? This will result in the value silently being zero - should we check this and fail if conversion fails here or are we happy that parsing may sometimes fail?

Reply:
Obviously not really great chances of errors here unless the API changes and then it would probably fail anyway. One might argue that a try for 0 price is a good idea here.